src/Microsoft.CodeAnalysis.Razor.Workspaces/DefaultTagHelperDescriptorFactory.cs

```diff
@@ -180,13 +217,21 @@ private IEnumerable<string> GetAllowedChildren(INamedTypeSymbol type, ErrorSink
 
             foreach (var name in allowedChildren)
             {
-                var valid = TryValidateName(
+                if (string.IsNullOrWhiteSpace(name))
+                {
+                    var whitespaceError = Workspaces.Resources.FormatTagHelperDescriptorFactory_InvalidRestrictChildrenAttributeNameNullWhitespace(
+                        nameof(TagHelperTypes.RestrictChildrenAttribute),
+                        tagHelperName);
+                    errorSink.OnError(SourceLocation.Zero, whitespaceError, length: 0);
+                }
+                else if (TryValidateName(
                     name,
-                    whitespaceError: "invalid",
-                    characterErrorBuilder: (invalidCharacter) => "invalid",
-                    errorSink: errorSink);
-
-                if (valid)
+                    invalidCharacter => Workspaces.Resources.FormatTagHelperDescriptorFactory_InvalidRestrictChildrenAttributeName(
+                        nameof(TagHelperTypes.RestrictChildrenAttribute),
```

Comment:
These bits are iffy but I wasn't sure if it was worth representing these values in the `TagHelperTypes`